commands/cli/parse.go

```diff
@@ -227,11 +229,61 @@ func parseArgs(inputs []string, stdin *os.File, argDefs []cmds.Argument, recursi
 		numInputs += 1
 	}
 
+	searchUnknownCmd := func(args []string) []string {
```

Comment:
maybe move this function out into its own file in the same directory, as well as the sorting code below.